### ISSUE TITLE
Mats event id and exon positions

### ISSUE THREAD

**Issue by nonesuch1936** [nuno-agostinho/alternativeSplicingEvents.hg38#1]
Hi, Looking the at MEE and SE events I find that the co-ordinates in Miso event Id do not match with the Exon start:end coordinates. Is there a reason for this? 
Also while trying to map coordinates using the Starsolo SJ.tab output there is no match with any junctions in the MEE or SE. 

Thanks
Suneetha

**Comment by nuno-agostinho** [comment]
Do you mean rMATS or Miso? Miso coordinates should be untouched, but the rMATS coordinates are altered to be comparable against other programs:

* 1 position is incremented to the start of AS events in the sense (`+`) strand
* 1 position is incremented to the end of AS events in the anti-sense (`-`) strand

If helpful, the positions are incremented in this part of the code:

https://github.com/nuno-agostinho/alternativeSplicingEvents.hg38/blob/70d8e99213408de90c25821552c6d5c175e68be6/inst/scripts/events.R#L114-L122

If you have other questions, feel free to ask!

Best,
Nuno

**Comment by nonesuch1936** [comment]
Hi, Thanks for your quick reply. I am referring to the MISO column. Eg. below. This is from the dataframe of Skipped events...

<img width="1188" alt="image" src="https://user-images.githubusercontent.com/91805205/138685734-575dc327-6a48-463e-a0b7-2e084dda171c.png">

The same is true for the MEE events also. Filtering for SJ events (formatted as Chrm:start:stop:strand does not give me any match all. Not sure which columns shd I use for filtering the SJ junctions. 

Best 
Suneetha

**Comment by nuno-agostinho** [comment]
Ah, right!

The most recent alternative splicing annotation that is available from MISO is for the [hg19 assembly](https://miso.readthedocs.io/en/fastmiso/annotation.html). I used the [liftOver](https://www.bioconductor.org/packages/release/workflows/html/liftOver.html) package to convert the coordinates from hg19 to hg38, but left the event IDs unchanged, so the coordinates in the IDs are actually the original hg19 coordinates.

In other words, you should use the columns **Constitutive exon 1 start/end**, **Alternative exon 1 start/end** and **Constitutive exon 2 start/end** to filter by splicing junctions.

Tell me if that was not clear and sorry for the confusion! :)

Best,
Nuno

**Comment by nonesuch1936** [comment]
Thanks for clearing that up! 

Best 
Suneetha

**Comment by nonesuch1936** [comment]
Hi Again,
    I have been trying to apply the annotations to the data to identify events but I  am not able to find any matches. The junctions were generated using StarSolo, and I have been trying to match junctions to known SE/MSE events in hg38.  I know there are SE events in the results because for  SE events we have an annotated file to filter with, but even these well-known SE events are not matched in the hgAnnotions38 file.  I figure I am doing something wrong, but have not been able to identify what.

**Comment by nuno-agostinho** [comment]
Hey @nonesuch1936, could you send me a sample of the StarSolo junctions to see if I understand what is going on? Maybe StarSolo coordinates are shifted by 1 position and that's why you don't have any matches?

I would suggest to check random skipped exon events in the annotation and try to see if nearby junctions from StarSolo exist to understand what is going on. The hg38 annotation does work with junction quantification data from [GTEx v8](https://gtexportal.org/home/index.html) and [recount2](https://jhubiostatistics.shinyapps.io/recount/), so I don't think the problem is with the annotation.
